int32 distance
---
bool status
